crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -609,3 +609,36 @@ async fn set_name() {
 
     room.set_name(name.to_owned()).await.unwrap();
 }
+
+#[async_test]
+async fn report_content() {
+    let (client, server) = logged_in_client().await;
+
+    let reason = "I am offended";
+    let score = int!(-80);
+
+    Mock::given(method("POST"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/report/\$offensive_event"))
+        .and(body_json(json!({
+            "reason": reason,
+            "score": score,
+        })))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EMPTY))
+        .expect(1)
```

Comment:
nit: if you're using `expect`, you need to call `server.verify().await` at the end of the test, for the expect to be checked.

Reply:
No we don't, it is checked automatically when the server is shutdown, and will panic. See the [docs for the method](https://docs.rs/wiremock/latest/wiremock/struct.Mock.html#method.expect).

Reply:
Ah indeed. We might have dreamed that in the Rust SDK team, because we thought at some point it was required :-)